crates/store/src/db/models/queries/insertions.rs

```diff
@@ -378,11 +379,22 @@ pub(crate) fn insert_notes(
     notes: &[(NoteRecord, Option<Nullifier>)],
 ) -> Result<usize, DatabaseError> {
     let count = diesel::insert_into(schema::notes::table)
-        .values(Vec::from_iter(
-            notes
-                .iter()
-                .map(|(note, nullifier)| NoteInsertRowRaw::from((note.clone(), *nullifier))),
-        ))
+        .values(Vec::from_iter(notes.iter().map(|(note, nullifier)| {
+            let stripped_details = note.details.as_ref().map(|details| {
+                let mut mast = details.script().mast().clone();
+                Arc::make_mut(&mut mast).strip_decorators();
+                let script = NoteScript::from_parts(mast, details.script().entrypoint());
+                let recipient =
+                    NoteRecipient::new(details.serial_num(), script, details.inputs().clone());
+                NoteDetails::new(details.assets().clone(), recipient)
+            });
+
+            let modified_note = NoteRecord {
+                details: stripped_details,
+                ..note.clone()
+            };
+            NoteInsertRowRaw::from((modified_note, *nullifier))
+        })))
```

Comment:
Would it be better to check if the notes had decorators before attempting to strip them as Bobbin suggested? Not sure what that would look like in terms of code but it does seem like we would save ourselves the note reconstruction, although maybe it doesn't make too much of a difference because we are cloning the note anyway.